src/_nebari/utils.py

```diff
@@ -280,3 +281,14 @@ def is_relative_to(self: Path, other: Path, /) -> bool:
 def set_do_environment():
     os.environ["AWS_ACCESS_KEY_ID"] = os.environ["SPACES_ACCESS_KEY_ID"]
     os.environ["AWS_SECRET_ACCESS_KEY"] = os.environ["SPACES_SECRET_ACCESS_KEY"]
+
+
+def set_azure_resource_group_name(
```

Comment:
I don't love the set_azure_resource_group_name since we aren't setting anything.  I'd just call it something along the lines of `azure_resource_group_name`, `build_azure_resource_group_name`, `construct_resource_group_name`, etc.